File: cpp/leetcode/294.cpp
class Solution {
public:
    bool canWin(string s) {
        if(s.empty()) return false;
        for(int i = 0; i < s.size() - 1; i++){
            if(s.substr(i, 2) == "++"){
                if(!canWin(s.substr(0, i) + "--" + s.substr(i+2))) return true;
            }
        }
        return false;
    }
};

File: python/leetcode/294.py
class Solution(object):  
    def canWin(self, s):  
        for i in range(len(s) - 1):  # try all possible ways
            if s[i:i+2] == "++":   
                current = s[0:i] + "--" + s[i+2:]
                if not self.canWin(current):  # if other people can win, then you win
                    return True
        return False  # no possible solution

